### PR TITLE
Api alerts

### PR DESCRIPTION
**Pull request by dani4571**
adding angularjs component for alerts
placing hx-alert elements in existing html to notify when api calls fail 
adding messaging service for broadcasting events which are currently picked up by alerts
removing duplicate searchAll function in api service

### CODE REVIEW

File: web/web/static/web/js/components/alerts.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+function AlertsController($scope, $filter, $log) {
+    var self = this;
+
+    self.$onInit = function() {
+        self.alerts = {
```
Comment (absalon-james) [inline web/web/static/web/js/components/alerts.js]: I think I would prefer alerts to be listed in order that they are received regardless of the type of alert. Can we make the component maintain just a list of alerts and move the type of alert into the alert object?

File: web/web/static/web/js/controllers/details.js [hunk 2]
```diff
@@ -55,7 +55,10 @@ angular.module('cloudSnitch').controller('DetailsController', ['$scope', 'cloudS
                 $scope.busy = false;
             }
         }, function(resp) {
-            // @TODO - Error handling
+            $log.error("In promise failure updateTimes");
```
Comment (absalon-james) [inline web/web/static/web/js/controllers/details.js]: Left over logging

File: web/web/static/web/js/controllers/details.js [hunk 3]
```diff
@@ -55,7 +55,10 @@ angular.module('cloudSnitch').controller('DetailsController', ['$scope', 'cloudS
...
+            messagingService.error("details_"+$scope.paneObj.paneIndex,
```
Comment (absalon-james) [inline web/web/static/web/js/controllers/details.js]: Spaces around concatenation operators

File: web/web/static/web/js/controllers/details.js [hunk 4]
```diff
@@ -76,6 +79,10 @@ angular.module('cloudSnitch').controller('DetailsController', ['$scope', 'cloudS
         ).then(function(result) {
             $scope.objectBusy = false;
         }, function(resp) {
+            $log.error("In promise failure updateObject");
```
Comment (absalon-james) [inline web/web/static/web/js/controllers/details.js]: Left over logging and spaces around operators.

File: web/web/static/web/js/controllers/details.js [hunk 5]
```diff
@@ -105,7 +112,10 @@ angular.module('cloudSnitch').controller('DetailsController', ['$scope', 'cloudS
         ).then(function(result) {
             $scope.children[childRef].busy = false;
         }, function(resp) {
-            // @TODO - error handling
+            $log.error("In promise failure searchChildren");
```
Comment (absalon-james) [inline web/web/static/web/js/controllers/details.js]: Left over logging and spaces around operators.

File: web/web/static/web/js/controllers/diff.js [hunk 6]
```diff
@@ -493,6 +493,10 @@ angular.module('cloudSnitch').controller('DiffController', ['$scope', '$interval
         }, function(resp) {
             stopPollingNodes();
             $scope.state = 'error';
+            $log.error("In promise failure getNodes");
+            messagingService.error("diff",
+                                   "API ERROR",
+                                   resp.status+" "+resp.statusText);
```
Comment (absalon-james) [inline web/web/static/web/js/controllers/diff.js]: Spaces around operators.

File: web/web/static/web/js/controllers/diff.js [hunk 7]
```diff
@@ -615,6 +619,10 @@ angular.module('cloudSnitch').controller('DiffController', ['$scope', '$interval
         }, function(resp) {
             stopPolling();
             $scope.state = 'error'
+            $log.error("In promise failure getStructure");
+            messagingService.error("diff",
+                                   "API ERROR",
+                                   resp.status+" "+resp.statusText);
```
Comment (absalon-james) [inline web/web/static/web/js/controllers/diff.js]: Spaces around operators.

File: web/web/static/web/js/controllers/pane.js [hunk 8]
```diff
@@ -102,7 +102,10 @@ angular.module('cloudSnitch').controller('ResultsController', ['$scope', 'cloudS
             $scope.count = data.count;
             $scope.busy = false;
         }, function(resp) {
-            console.log("error searching page.");
+            $log.error("In promise failure searchPage");
+            messagingService.error("searchpage_"+$scope.paneObj.paneIndex,
+                                   "API ERROR",
+                                   resp.status+" "+resp.statusText);
```
Comment (absalon-james) [inline web/web/static/web/js/controllers/pane.js]: Spaces around operators.

File: web/web/static/web/js/controllers/reporting.js [hunk 9]
```diff
@@ -106,6 +106,10 @@ angular.module('cloudSnitch').controller('ReportingController', ['$scope', 'repo
         }, function(resp) {
             $scope.serverErrors = resp.data;
             $scope.busy = false;
+            $log.error("In promise failure submitReport");
+            messagingService.error("reporting",
+                                   "API ERROR",
+                                   resp.status+" "+resp.statusText);
```
Comment (absalon-james) [inline web/web/static/web/js/controllers/reporting.js]: Spaces around operators.

File: web/web/static/web/js/services/api.js [hunk 10]
```diff
@@ -270,68 +270,6 @@ angular.module('cloudSnitch').factory('cloudSnitchApi', ['$http', '$q', 'timeSer
         });
     };
 
-    service.searchAll = function(model, identity, time, filters, sink) {
```
Comment (absalon-james) [inline web/web/static/web/js/services/api.js]: Good find.

File: web/web/static/web/js/services/reports.js [hunk 11]
```diff
@@ -21,8 +21,11 @@ angular.module('cloudSnitch').factory('reportsService', ['$rootScope', 'cloudSni
                 service.reportsMap[service.reports[i].name] = service.reports[i];
             }
             $rootScope.$broadcast('reports:update');
-        }, function(error) {
-            // @TODO - Do something with error
+        }, function(resp) {
+            $log.error("In promise failure updateReports");
+            messagingService.error("master_alert",
+                                   "API ERROR",
+                                   resp.status+" "+resp.statusText);
```
Comment (absalon-james) [inline web/web/static/web/js/services/reports.js]: Spaces around operators.

File: web/web/static/web/js/services/types.js [hunk 12]
```diff
@@ -72,8 +72,11 @@ angular.module('cloudSnitch').factory('typesService', ['cloudSnitchApi', functio
         cloudSnitchApi.paths().then(function(result) {
             service.paths = result;
             service.pathsLoading = false;
-        }, function(error) {
-            // @TODO - Do something with errors.
+        }, function(resp) {
+            $log.error("In promise failure updatePaths");
+            messagingService.error("master_alert",
+                                   "API ERROR",
+                                   resp.status+" "+resp.statusText);
```
Comment (absalon-james) [inline web/web/static/web/js/services/types.js]: Spaces around operators.

File: web/web/static/web/js/services/types.js [hunk 13]
```diff
@@ -87,8 +90,11 @@ angular.module('cloudSnitch').factory('typesService', ['cloudSnitchApi', functio
             for (var i = 0; i < service.types.length; i++) {
                 service.typeMap[service.types[i].label] = service.types[i];
             }
-        }, function(error) {
-            // @TODO - Do something with error
+        }, function(resp) {
+            $log.error("In promise failure updateTypes");
+            messagingService.error("master_alert",
+                                   "API ERROR",
+                                   resp.status+" "+resp.statusText);
```
Comment (absalon-james) [inline web/web/static/web/js/services/types.js]: Spaces around operator.

File: web/web/static/web/html/alert.html [hunk 14]
```diff
@@ -0,0 +1,14 @@
+<div ng-repeat="(level, alerts) in $ctrl.alerts" >
```
Comment (absalon-james) [inline web/web/static/web/html/alert.html]: I would prefer to see alerts in the order that they arrive.

**Comment by dani4571** [comment]
Updates for formatting and list added. Alert view for site wide alerts coming in follow up PR

**Pull request closed without merging**